Makefile

```diff
@@ -23,65 +57,40 @@ endif
-	validate_param() { \
-		param_name="$${1}"; param_value="$${2}"; error_msg="$${3}"; \
-		[ -n "$${param_value}" ] || { echo "Error: $${error_msg}"; echo "$${usage_msg}"; exit 1; }; \
-	}; \
-	validate_param "SDK_IMAGE" "$(SDK_IMAGE)" "SDK_IMAGE parameter is required"; \
-	validate_param "KVER" "$(KVER)" "KVER parameter is required"; \
-	echo "All parameter checks passed. Merging kernel configs for version $(KVER)..."
-
+	@test $${SDK_IMAGE?not set!}
+	@test $${KVER?not set!}
+	@test $${RPM_FILE?not set!}
+	$(eval KERNEL_PACKAGE_DIR := $(TOP)packages/kernel-$(KVER))
+	@test -d "$(KERNEL_PACKAGE_DIR)"
+	@echo "Merging kernel configs for version $(KVER)..."
+	@echo "Using SDK image: $(SDK_IMAGE)"
+	@echo "Using RPM file: $(RPM_FILE)"
 	@tmpdir=$$(mktemp -d); \
-	if [ $$? -ne 0 ]; then \
-		echo "Error: Failed to create temporary directory"; \
-		exit 1; \
-	fi; \
+	trap "rm -rf $${tmpdir}" EXIT; \
 	cp "$(RPM_FILE)" "$${tmpdir}/kernel-source.rpm"; \
-	if [ $$? -ne 0 ]; then \
-		echo "Error: Failed to copy RPM file to temporary directory"; \
-		rm -rf "$${tmpdir}"; \
-		exit 1; \
-	fi; \
-	cd packages/kernel-$(KVER) && \
+	cd $(KERNEL_PACKAGE_DIR) && \
 	docker run --rm \
-		-v "$(PWD)/packages/kernel-$(KVER)/":/kernel-package \
+		-v "$(KERNEL_PACKAGE_DIR)/":/kernel-package \
 		-v "$${tmpdir}":/work \
-		-v "$(PWD)/packages/microcode/":/microcode \
-		-v "$(PWD)/tools/latest-kernel-full-config.sh":/latest-kernel-full-config.sh \
+		-v "$(TOP)packages/microcode/":/microcode \
+		-v "$(TOP)tools/latest-kernel-full-config.sh":/latest-kernel-full-config.sh \
 		--user "$$(id -u):$$(id -g)" \
 		--name "kernel-$(KVER)-inner-full" \
 		"$(SDK_IMAGE)" \
-		bash -c "source /latest-kernel-full-config.sh && inner_full_config"; \
-	docker_exit_code=$$?; \
-	rm -rf "$${tmpdir}"; \
-	if [ $$docker_exit_code -ne 0 ]; then \
-		echo "Error: Docker container execution failed with exit code $$docker_exit_code"; \
-		exit $$docker_exit_code; \
-	fi; \
-	echo "Successfully merged kernel configs for version $(KVER)"
+		bash -c "source /latest-kernel-full-config.sh && inner_full_config"
+	@echo "Successfully merged kernel configs for version $(KVER)"
 
 full-config:
-	@if [ -z "$(RPM_FILE)" ]; then \
-		echo "Error: RPM_FILE parameter is required"; \
-		echo "Usage: make full-config RPM_FILE=/path/to/kernel-source.rpm"; \
-		exit 1; \
-	fi; \
-	echo "Checking kernel configuration dependencies..."; \
-	source $(KERNEL_CONFIG_SCRIPT) && check_dependencies; \
-	echo "All dependencies are available."; \
-	rpm_file=$$(realpath "$(RPM_FILE)"); \
-	if [ ! -f "$${rpm_file}" ]; then \
-		echo "Error: RPM file not found: $${rpm_file}"; \
-		exit 1; \
-	fi; \
-	sdk_image=$$(source $(KERNEL_CONFIG_SCRIPT) && resolve_bottlerocket_sdk); \
+	@test $${RPM_FILE?not set!}
+	@echo "Checking kernel configuration dependencies..."
+	$(eval MISSING_TOOLS := $(shell for tool in rpm2cpio cpio tar docker tq; do command -v $$tool >/dev/null 2>&1 || echo $$tool; done))
```

Comment:
You are doing your validation on at least `tq`, after you have already attempted to use it in the first section of your Makefile.